### DISCUSSION TITLE
Simple question

### DISCUSSION THREAD

**Discussion by mrkrausestem** [community/community#127577] [GitHub Education]
### Hi everyone,

I have been using classroom for several years now. I have relied on github desktop for the students to clone and checkin their assignments. Is there a web based solution that I can use instead? The students do not have access to terminal on their macs to use the CLI and my tech people do not want to deploy desktop either. I am unaware of any other method for checking in code.

Help or confirmation of my fears would be helpful.

**Comment by OG-GY** [comment]
There are indeed web-based solutions that can help you manage your classroom's coding assignments without requiring students to use GitHub Desktop or the terminal. Here are a few popular options:

### 1. **GitHub Classroom**

GitHub Classroom is an excellent tool designed specifically for educational purposes. It integrates seamlessly with GitHub and provides a web-based interface for both instructors and students. Here’s what it offers:
- **Assignment Distribution**: You can create and distribute assignments as GitHub repositories.
- **Automated Grading**: Integrate with GitHub Actions for automated grading.
- **Web-based Code Editing**: Students can use GitHub's web editor to make changes directly in the browser.

### 2. **Repl.it**

Repl.it is an online IDE that supports many programming languages and provides a collaborative environment.
- **Assignment Management**: You can create classrooms, distribute assignments, and track student progress.
- **Web-based Coding**: Students can write, run, and debug code directly in their browser.
- **Git Integration**: You can link Repl.it projects with GitHub repositories for version control.

### 3. **GitPod**

GitPod provides a complete development environment in the cloud, accessible via a web browser.
- **GitHub Integration**: You can connect GitPod directly to GitHub repositories.
- **IDE Features**: It offers a fully-featured IDE experience within the browser.
- **Persistent Workspaces**: Students can save their progress and resume later.

### 4. **Glitch**

Glitch is another web-based platform focused on collaborative coding, especially for web development projects.
- **Easy Collaboration**: It allows real-time collaboration on coding projects.
- **Version Control**: Though not as robust as GitHub, it provides basic version control features.
- **Web-based Editing**: Students can edit code directly in their browser.

### 5. **Google Colab**

For Python-based assignments, Google Colab is a great option.
- **Notebook Interface**: It's a Jupyter notebook environment accessible through the web.
- **Git Integration**: While primarily for Python, it has Git integration features.
- **Code and Document**: Students can write and execute code, and document their process within the same environment.

### Implementation Tips

- **Onboarding and Training**: Whichever platform you choose, ensure that you provide a tutorial or training session for your students to get familiar with the new tool.
- **Support Resources**: Have resources or support available for students who might face technical issues.
- **Integration with LMS**: If your school uses a Learning Management System (LMS), check if the chosen tool integrates well with it for smoother workflow.

These solutions should help alleviate the concerns of not having terminal access or GitHub Desktop, providing a smooth and effective way for students to submit and manage their coding assignments online.
